### PR TITLE
First version of n3c extractor using pyspark

### PR DESCRIPTION
**Pull request by smitalimaye**

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+cdm_database_schema: gcp-project-id.deid-dataset-name
+results_database_schema: gcp-project-id.cohorts-dataset-name
+# make sure all cdm tables are created and populated
+persist_database_schema: gcp-project-id.subset-dataset-name
+
+# pyspark configurations
+wait_for_job_completion_in_min: 120
+
+# configuration for pyspark batch to use
+extract_config_file: config.yaml
```
Comment (smitalimaye) [inline config.yaml]: This can  be removed from here, we can use the value supplied as an argument

File: src/n3c-extractor/spark_extractor.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+spark_bq_jar = 'gs://spark-lib/bigquery/spark-bigquery-with-dependencies_2.12-0.27.1.jar'
+gs_bucket_config = 'gcs_bucket'
+prefix_config = 'prefix'
+cdm_tables_config = 'cdm_tables'
+data_counts_folder = 'data_counts'
+
+timestamp_format = 'yyyy-MM-dd HH:mm:ss'
+mainfest = 'manifest'
+
+class SparkSqlBatch:
```
Comment (smitalimaye) [inline src/n3c-extractor/spark_extractor.py]: Should this go to another file as plain script?

File: src/n3c-extractor/spark_extractor.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    bucket = storage_client.get_bucket(self.gcs_bucket)
+    csv_folder = 'data_counts'
+    if self.prefix != None:
+      csv_folder = f'{self.prefix}/data_counts'
+    blob = bucket.blob(f'{csv_folder}/header.csv')
+    blob.upload_from_string('"table_name"|"row_count"\n')
+
+    print('Done extraction data_counts/header file')
+
+class PysparkBatchRunner:
```
Comment (smitalimaye) [inline src/n3c-extractor/spark_extractor.py]: Should this be a module?

File: src/n3c-extractor/spark_extractor.py [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+              output_folder = datafiles_dir
+            file = f'{output_folder}/{cdm_table_name_upper}'
+            content = bucket.blob(blob.name)
+            content.download_to_filename(file)
+            print(f'Downaloaded {output_dir}{cdm_table_name_upper}')
+
+      print('Done downloading csvs...')
+      
+  
+def main():
```
Comment (smitalimaye) [inline src/n3c-extractor/spark_extractor.py]: Should this go to `__init__.py` file?

File: pyproject.toml [hunk 5]
```diff
@@ -150,5 +150,25 @@ package-data = {"sample" = ["*.dat"]}
 [build-system]
 # These are the assumed default build requirements from pip:
 # https://pip.pypa.io/en/stable/reference/pip/#pep-517-and-518-support
-requires = ["setuptools>=43.0.0", "wheel"]
+requires = ["setuptools>=43.0.0", "wheel", "google-cloud-dataproc", "pyyaml", "gcsfs"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "n3c_spark_extractor"
```
Comment (smitalimaye) [inline pyproject.toml]: Does this name look right? It doesn't look cool though :)

File: pyproject.toml [hunk 6]
```diff
@@ -150,5 +150,25 @@ package-data = {"sample" = ["*.dat"]}
 [build-system]
 # These are the assumed default build requirements from pip:
 # https://pip.pypa.io/en/stable/reference/pip/#pep-517-and-518-support
-requires = ["setuptools>=43.0.0", "wheel"]
+requires = ["setuptools>=43.0.0", "wheel", "google-cloud-dataproc", "pyyaml", "gcsfs"]
```
Comment (smitalimaye) [inline pyproject.toml]: This may need some more updates

**Comment by smitalimaye** [comment]
This is ready to merge

**Review by jmesterh — APPROVED** [review]

**Pull request merged**